tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Settings/PartialViews.spec.ts

```diff
@@ -0,0 +1,240 @@
+    expect(await umbracoApi.partialView.nameExistsAtRoot(partialViewFileName)).toBeFalsy;
+    // TODO: when frontend is ready, uncomment this to verify the old partial view name is NOT displayed under the Partial Views section
+    //await expect(page.locator('umb-tree-item', {hasText: 'Partial Views'}).getByLabel(partialViewFileName)).toHaveCount(0);
+    
+    // Clean
+    await umbracoApi.partialView.ensureNameNotExistsAtRoot(updatedPartialViewFileName);
+  });
+
+  test('can update a partial view content', async ({page, umbracoApi}) => {
+    const partialViewName = 'TestPartialView';
+    const partialViewFileName = partialViewName + ".cshtml";
+    const updatedPartialViewContent = '@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage\r\n' +
+      '@{\r\n' +
+      '\tLayout = null;\r\n' +
+      '}\r\n' +
+      '<p>AcceptanceTests</p>';
+
+    // Arrange
+    await umbracoApi.partialView.ensureNameNotExistsAtRoot(partialViewFileName);
+    await umbracoApi.partialView.create(partialViewFileName, "@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage\r\n", "/");
+
+    //Act
+    gotoPartialViewDetail(page, partialViewFileName);
+    await page.locator('textarea.inputarea').clear();
+    await page.locator('textarea.inputarea').fill(updatedPartialViewContent);
+    await page.getByLabel('Save').click();
+
+    // Assert
+    const updatedPartialView = await umbracoApi.partialView.getByNameAtRoot(partialViewFileName);
+    expect(updatedPartialView.content).toBe(updatedPartialViewContent);
+    
+    // Clean
+    await umbracoApi.partialView.ensureNameNotExistsAtRoot(partialViewFileName);
+  });
+
+  test('can use query builder for a partial view', async ({page, umbracoApi, umbracoUi}) => {
+    const partialViewName = 'TestPartialView';
+    const partialViewFileName = partialViewName + ".cshtml";
+    const expectedTemplateContent = '\r\n' +
+    '@{\r\n' +
+    '\tvar selection = Umbraco.ContentAtRoot().FirstOrDefault()\r\n' +
+    '    .Children()\r\n' +
+    '    .Where(x => x.IsVisible());\r\n' +
+    '}\r\n' +
+    '<ul>\r\n' +
+    '\t@foreach (var item in selection)\r\n' +
+    '\t{\r\n' +
+    '\t\t<li>\r\n' +
+    '\t\t\t<a href="@item.Url()">@item.Name()</a>\r\n' +
+    '\t\t</li>\r\n' +
+    '\t}\r\n' +
+    '</ul>\r\n' +
+    '\r\n' +
+    '@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage\r\n';
+
+
+    // Arrange
+    await umbracoApi.partialView.ensureNameNotExistsAtRoot(partialViewFileName);
+    await umbracoApi.partialView.create(partialViewFileName, "@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage\r\n", "/");
+
+    // Act
+    gotoPartialViewDetail(page, partialViewFileName);
+    await page.locator('#query-builder-button').getByLabel('Query builder').click();
+    await page.waitForTimeout(1000);
+    await page.locator('#property-alias-dropdown').getByLabel('Property alias').click({force: true});
+    await page.waitForTimeout(1000);
```

Comment:
Is there a reason for these two timeouts? 🤔

Reply:
I use another way to wait element instead of using waitTimeout. Please re-check it.